### ISSUE TITLE
FIM false positive caused by non case sensitive comparison

### ISSUE THREAD

**Issue by j0serca** [wazuh/wazuh#15800]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.3.10 | Syscheck/FIM | Agent | Packages/Sources | Windows |

# Bug Description

Hi team,
Some FIM false positives have been reported,  specifically they get permissions changes like this one:

`syscheck.win_perm_before:`
```
{
  "allowed": [
    "DELETE",
    "READ_CONTROL",
    "WRITE_DAC",
    "WRITE_OWNER",
    "SYNCHRONIZE",
    "READ_DATA",
    "WRITE_DATA",
    "APPEND_DATA",
    "READ_EA",
    "WRITE_EA",
    "EXECUTE",
    "READ_ATTRIBUTES",
    "WRITE_ATTRIBUTES"
  ],
  "name": "Domain Admins"
},
```

`syscheck.win_perm_after:`
```
{
  "allowed": [
    "DELETE",
    "READ_CONTROL",
    "WRITE_DAC",
    "WRITE_OWNER",
    "SYNCHRONIZE",
    "READ_DATA",
    "WRITE_DATA",
    "APPEND_DATA",
    "READ_EA",
    "WRITE_EA",
    "EXECUTE",
    "READ_ATTRIBUTES",
    "WRITE_ATTRIBUTES"
  ],
  "name": "domain admins"
},
```

As you see, the only modification is the group name, which was all in lowercase. The original one was never modified and this "new" group does not exist on AD, and consequently, has never been used.
It seems that the comparison between old and new permissions is case-sensitive, causing these false positives (I'm not sure why Windows generates these differences in the group's names). It'd be ideal to not use a case-sensitive comparison on this case.

Thank you!

**Comment by jotacarma90** [comment]
After investigating this problem I have deduced that the problem occurs at two points in the FIM code:
- When we calculate the checksum of the events, in fim_get_checksum, we do it using the string associated to the extracted permissions, it is case sensitive.
- When we compare the strings, in fim_json_compare_attrs, we do a strcmp between the new and old permissions, and this is also case sensitive.

The change that could solve this problem is to convert the account name to lowercase as soon as it is extracted. I have created a PR draft with the minifix:
https://github.com/wazuh/wazuh/pull/15843

However, we have not yet been able to find a way to reproduce this error in a local Windows environment.
